Assertion failure in tests/charm++/sdag on pami/pamilrts

Autobuild log: http://charm.cs.illinois.edu/autobuild/old.2019_07_25__01_04/pamilrts-bluegeneq.txt

```
2019-07-25 10:10:36.189 (INFO ) [0xfffb646cb70] 46295:tatu.runjob.client: scheduler job id is 732601
2019-07-25 10:10:36.193 (INFO ) [0xfff99ee37b0] 46295:tatu.runjob.monitor: monitor started
2019-07-25 10:10:38.493 (INFO ) [0xfff99ee37b0] 46295:tatu.runjob.monitor: task record 2243792 created
2019-07-25 10:10:38.494 (INFO ) [0xfffb646cb70] VST-00660-11771-32:46295:ibm.runjob.client.options.Parser: set local socket to runjob_mux from properties file
2019-07-25 10:10:40.860 (INFO ) [0xfffb646cb70] VST-00660-11771-32:2412441:ibm.runjob.client.Job: job 2412441 started
2019-07-25 10:10:43.161 (INFO ) [0xfff99ee37b0] 46295:tatu.runjob.monitor: tracklib completed
[0] Assertion "cb_sum == expected_cb_sum" failed in file refnum.ci line 132.
------------- Processor 0 Exiting: Called CmiAbort ------------
Reason: Assertion "cb_sum == expected_cb_sum" failed in file refnum.ci line 132.
sdag_tests: machine.C:816: void LrtsAbort(const char *): Assertion `0' failed.
2019-07-25 10:11:15.685 (WARN ) [0xfffb646cb70] VST-00660-11771-32:2412441:ibm.runjob.client.Job: terminated by signal 6
2019-07-25 10:11:15.685 (WARN ) [0xfffb646cb70] VST-00660-11771-32:2412441:ibm.runjob.client.Job: abnormal termination by signal 6 from rank 0
2019-07-25 10:11:15.685 (INFO ) [0xfffb646cb70] tatu.runjob.client: task terminated by signal 6
2019-07-25 10:11:15.686 (INFO ) [0xfff99ee37b0] 46295:tatu.runjob.monitor: monitor terminating
2019-07-25 10:11:15.690 (INFO ) [0xfffb646cb70] tatu.runjob.client: monitor completed
make[3]: *** [test] Error 134
make[2]: *** [test-sdag] Error 2
```

--- a/tests/charm++/sdag/refnum.ci
+++ b/tests/charm++/sdag/refnum.ci
@@ -77,7 +77,7 @@ module refnum {
             }
             when recvData[index](CkDataMsg* m) serial {
               refnum_sum += index;
-              cb_sum += (int)m->data[0];
+              cb_sum += ((int*)(m->getData()))[0];
               delete m;
             }
           }
@@ -97,7 +97,7 @@ module refnum {
           }
           when recvData[index](CkDataMsg* m) serial {
             refnum_sum += index;
-            cb_sum += (int)m->data[0];
+            cb_sum += ((int*)(m->getData()))[0];
             delete m;
           }
         }
@@ -115,7 +115,7 @@ module refnum {
           }
           when recvData[index](CkDataMsg* m) serial {
             refnum_sum += index;
-            cb_sum += (int)m->data[0];
+            cb_sum += ((int*)(m->getData()))[0];
             delete m;
           }
         }

